src/main/java/com/datadoghq/sketch/ddsketch/store/DenseStore.java

```diff
@@ -322,16 +324,19 @@ public Bin next() {
     }
 
     @Override
-    public com.datadoghq.sketch.ddsketch.proto.Store toProto() {
-        final com.datadoghq.sketch.ddsketch.proto.Store.Builder builder =
-            com.datadoghq.sketch.ddsketch.proto.Store.newBuilder();
+    public int serializedSize() {
         if (counts != null) {
-            // For the dense store, we use the dense representation to encode the bin counts.
-            builder.setContiguousBinIndexOffset(minIndex);
-            for (long index = minIndex; index <= maxIndex; index++) {
-                builder.addContiguousBinCounts(counts[(int) index - offset]);
-            }
+            return Serializer.sizeOfCompactDoubleArray(2, maxIndex - minIndex + 1)
+                    + Serializer.signedIntFieldSize(3, minIndex);
+        }
+        return 0;
+    }
+
+    @Override
+    public void serialize(Serializer serializer) {
+        if (counts != null) {
+            serializer.writeCompactArray(2, counts, minIndex - offset, maxIndex - minIndex + 1);
+            serializer.writeSignedInt32(3, minIndex);
```

Comment:
That's a detail, but I realize we should probably have had the `minIndex` first to allow optimizations based on the value of `minIndex` that would have allowed stopping deserializing early.

Reply:
ok makes sense

Reply:
Although, this is pretty dangerous to rely on

Reply:
Yes. I think it's not worth changing at this point, especially since other libs rely on the current protobuf message format.

Reply:
We can put it first anyway to help partial deserialisers. The keys don't need to be ordered in the message, they just need to correspond to the field numbers in the schema. 

Incidentally, for non-repeated fields, the parser is obliged to take the _last_ value matching the key; the wire format permits repeating non-repeated fields. So arguably every partial protobuf deserialisation technique I've seen is wrong, no matter how well it can work if the circle of trust is small enough.

Reply:
I see, up to you then, but I think that makes partial deserialization optimizations less interesting then, and the non-dependency on the order in the message means that we can leave that for later if the need for it really arises.

Reply:
this would make testing painful though because it breaks comparisons of the binary output, let's address it another time.